functions/index.js

```diff
@@ -1,9 +1,65 @@
 const functions = require('firebase-functions')
+const axios = require('axios')
+const mailchimp = require('@mailchimp/mailchimp_marketing')
 
-// Create and Deploy Your First Cloud Functions
-// https://firebase.google.com/docs/functions/write-firebase-functions
+exports.processCalendlyEventInvitee = functions.https.onRequest(async (request, resp) => {
+  const { calendlyInviteeURI } = request.body
+  const token = 'eyJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczovL2F1dGguY2FsZW5kbHkuY29tIiwiaWF0IjoxNjE4NjA3MDM3LCJqdGkiOiIxZjliMjRjOC03NGEwLTRlNzMtOGMxMy0xMGI0MjNlYmI2NjQiLCJ1c2VyX3V1aWQiOiJHR0hBUVVBQzNONUxUUlgyIiwib3JnX3V1aWQiOiJESENIVVJEQTZZWEJRUzZMIn0.UxvkeOZGkCiq_9RE6bU2PV-mD0keGKE7rwuvOZ2SZLY'
+  const listId = 'bfb4a12429'
 
-exports.addMemberToMailchimp = functions.https.onRequest((request, response) => {
-  functions.logger.info('Hello logs!', { structuredData: true })
-  response.send('OK')
+  resp.set('Access-Control-Allow-Origin', '*')
+  if (request.method === 'OPTIONS') {
+    resp.set('Access-Control-Allow-Methods', 'POST')
+    resp.set('Access-Control-Allow-Headers', 'Content-Type')
+    resp.set('Access-Control-Max-Age', '3600')
+    resp.status(204).send('')
+    return undefined
+  }
+
+  mailchimp.setConfig({
+    apiKey: '9ab820bcffa760d9f787083c910ddf6e-us1',
+    server: 'us1'
+  })
+
+  if (calendlyInviteeURI.startsWith('https://api.calendly.com/')) {
+    const info = await getUser()
+    const subscribingUser = {
+      firstName: info.body.name,
+      lastName: '',
+      email: info.body.email
+    }
+    addsubscribingUser(listId, subscribingUser)
+  } else {
+    throw new Error('401 Unauthorized')
```

Comment:
You need to send the response to the client using the `resp` variable that you received.
- You do this once you finish doing everything you need to do
- If everything went OK, you must answer with a `200` status code: `resp.status(200).end()`
- If an error occurred (as in this case), you can send an appropriate status code and optionally a message: `resp.status(401).end()` or `resp.status(401).send('Unauthorized')`